clients/filesystem-hadoop3/build.gradle.kts

```diff
@@ -14,6 +14,7 @@ dependencies {
   implementation(project(":clients:client-java-runtime", configuration = "shadow"))
   implementation(libs.caffeine)
 
+  testCompileOnly(project(":core")) // fix error-prone compile error
```

Comment:
I think the server-common has already introduced the core project. But there is still a compile error.

Reply:
@coolderli what is the error here?

Reply:
```
错误: 无法访问Configs
  找不到com.datastrato.gravitino.Configs的类文件
  有关详细信息, 请参阅以下堆栈跟踪。
  com.sun.tools.javac.code.Symbol$CompletionFailure: 找不到com.datastrato.gravitino.Configs的类文件
```

Reply:
It's weird that gvfs needs this class in `core`.

Reply:
In TestOauth2Client, it use Config to init ServerAuthenticator.

Reply:
I think we need to fix this dependency issue, not add it to the dependency to make it work. Ideally we should not rely on core.

Reply:
@coolderli I did a slight change to avoid relying on this, can you please help to check?

Reply:
LGTM. It's reasonable.